tests/e2e-instrumentation-signatures/e2e-containers_data_source.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/detect"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type e2eContainersDataSource struct {
+	cb             detect.SignatureHandler
+	containersData detect.DataSource
+}
+
+func (sig *e2eContainersDataSource) Init(ctx detect.SignatureContext) error {
+	sig.cb = ctx.Callback
+	containersData, ok := ctx.GetDataSource("tracee", "containers")
+	if !ok {
+		return fmt.Errorf("containers data source not registered")
+	}
+	if containersData.Version() > 1 {
```

Comment:
good reason for the version number suggestion (ditto in other comment).

Reply:
Could you give a pseudo code example to explain how it would be an improvement?

Reply:
no pseudocode needed, sorry, its literally just an integer value being manually monotonically increased each time the datasource value "interface{}" is altered (like in the given example, from the DNS RFC document, it is basically it).

Reply:
TODO: check if you want to address this after freeze (freezing tomorrow). This will be merged as is by https://github.com/aquasecurity/tracee/pull/3142